framework/src/governance/lib/datazone-authorizer/resources/msk-producer-authorizer/index.mjs

```diff
@@ -0,0 +1,55 @@
+import { KafkaClient, UpdateConnectivityCommand, PutClusterPolicyCommand } from "@aws-sdk/client-kafka"
+export const handler = async(event) => {
+    const client = new KafkaClient()
+    const detail = event.detail
+
+    if (detail.eventType === "SUBSCRIBED") {
+        const asset = detail.asset
+        const environments = detail.environments
+
+        const kafkaForm = asset.formOutput.find((form) => form.typeName === "KafkaForm" )
+        const { clusterArn } = JSON.parse(kafkaForm.content)
+
+        await client.send(new UpdateConnectivityCommand({
```

Comment:
Why do we have this here? We should not just trigger the action without verifying if its was successfully applied. In addition it mutate the infrastructure (the msk cluster), ideally we should have the user perform this action outside the authorizer.

Reply:
file removed